flink-runtime/src/main/java/org/apache/flink/runtime/rpc/RpcEndpoint.java

```diff
@@ -323,6 +355,15 @@ public void validateRunsInMainThread() {
 		assert MainThreadValidatorUtil.isRunningInExpectedThread(currentMainThread.get());
 	}
 
+	/**
+	 * Rpc endpoint's state.
+	 */
+	public enum State {
```

Comment:
If we only expose `isRunning`, then we could simplify the state machine to `boolean isRunning` which is set to `true` in `doStart` and to `false` in `doStop`.